hassio/src/dialogs/snapshot/dialog-hassio-snapshot-upload.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  CSSResult,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { fireEvent } from "../../../../src/common/dom/fire_event";
+import { createCloseHeading } from "../../../../src/components/ha-dialog";
+import { HassDialog } from "../../../../src/dialogs/make-dialog-manager";
+import { haStyleDialog } from "../../../../src/resources/styles";
+import type { HomeAssistant } from "../../../../src/types";
+import "../../components/hassio-upload-snapshot";
+import { HassioSnapshotUploadDialogParams } from "./show-dialog-snapshot-upload";
+
+@customElement("dialog-hassio-snapshot-upload")
+export class DialogHassioSnapshotUpload extends LitElement
+  implements HassDialog {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @internalProperty() private _params?: HassioSnapshotUploadDialogParams;
+
+  public async showDialog(
+    params: HassioSnapshotUploadDialogParams
+  ): Promise<void> {
+    this._params = params;
+    await this.updateComplete;
+  }
+
+  public closeDialog(): void {
+    this._params!.loadData();
+    this._params = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render(): TemplateResult {
+    if (!this._params) {
+      return html``;
+    }
+
+    return html`
+      <ha-dialog
+        open
+        scrimClickAction
+        escapeKeyAction
+        hideActions
+        @closed=${this.closeDialog}
+        .heading=${createCloseHeading(this.hass, "Upload snapshot")}
+      >
+        <hassio-upload-snapshot
+          @snapshot-uploaded=${this.closeDialog}
```

Comment:
Can we show the detail dialog of the uploaded snapshot after upload is completed?

Reply:
Yes, do we want to?

Reply:
I guess the reason for uploading would be to restore it?

Reply:
Seems too logical :smiley:

Reply:
This also makes a lot of sense if added elsewhere (like during onboarding) :+1: